### ISSUE TITLE
`tailsrv` crashes when remote closes connection

### ISSUE THREAD

**Issue by Fuuzetsu** [asayers/tailsrv#8]
1. serve some random file:

```shell
$ tailsrv -p 6666 ~/ordercaches/Kospi200/v121/20160908.ordercache.zst
```

2. Start following
```shell
echo 0 | nc localhost 6666
```

3. Hit Ctrl-C on the `nc` process, causing a connection drop.

4. Crash

```
2024-12-13T07:27:03.404675Z  INFO tailsrv: Set up the io_uring
2024-12-13T07:27:03.404706Z  INFO tailsrv: Created an eventfd fd=4
2024-12-13T07:27:03.404716Z  INFO tailsrv: Polling the eventfd for events
2024-12-13T07:27:03.404756Z  INFO tailsrv: Bound socket listen_addr=0.0.0.0:6666
2024-12-13T07:27:03.405081Z  INFO {path=/home/shana/ordercaches/Kospi200/v121/20160908.ordercache.zst}: tailsrv: Opened file
2024-12-13T07:27:03.405119Z  INFO tailsrv: Initial file size: 268137 kiB
2024-12-13T07:27:03.405138Z  INFO tailsrv: Registered file with the io_uring file_fd=Fixed(0)
2024-12-13T07:27:03.405170Z  INFO tailsrv: Created an inotify watch path=/home/shana/ordercaches/Kospi200/v121/20160908.ordercache.zst fd=7
2024-12-13T07:27:03.405181Z  INFO tailsrv: Polling the inotify watch for events
2024-12-13T07:27:03.405188Z  INFO tailsrv: Starting runloop
2024-12-13T07:27:10.278061Z  INFO {client_id=49672}: tailsrv: Connected
2024-12-13T07:27:10.278119Z  INFO {client_id=49672}: tailsrv: Starting from initial offset 0
thread 'main' panicked at src/main.rs:252:17:
assertion `left == right` failed
  left: 65536
 right: 12032
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Crash is here: https://github.com/asayers/tailsrv/blob/48c60885a6e6e36d1364670dc6d6d57bf7f77ba4/src/main.rs#L252

**Comment by asayers** [comment]
Oof. I'm so surprised I didn't hit this during my own testing!

**Comment by asayers** [comment]
~I'm having trouble reproducing.  What version is your kernel?~

EDIT: Nevermind, I got it.  It just doesn't happen every time.

**Comment by asayers** [comment]
Published v0.9.2 which fixes this bug.
